calicoctl/commands/node/run.go

```diff
@@ -192,12 +197,24 @@ Description:
 
 	// Create a mapping of environment variables to values.
 	envs := map[string]string{
-		"NODENAME": name,
+		"NODENAME":                  name,
 		"CALICO_NETWORKING_BACKEND": backend,
 		"NO_DEFAULT_POOLS":          noPoolsString,
 		"CALICO_LIBNETWORK_ENABLED": fmt.Sprint(!disableDockerNw),
 	}
 
+	// Validate the ifprefix to only allow alphanumeric characters
+	ifprefixMatch := regexp.MustCompile(`^[a-zA-Z0-9]*$`)
```

Comment:
Can you declare this as a const or var at the top - it'll be easier to spot it and change it later if we need to change the regex.